.github/workflows/windows-msys2.yml

```diff
@@ -78,93 +81,89 @@ jobs:
             echo CFGOPT="--with-pkgversion=GnuCOBOL-CI-MSYS2" >> $env:GITHUB_ENV
-          } else {
+          } Else {
             echo DISTDIR=GnuCOBOL_mingw_dbg >> $env:GITHUB_ENV
-            echo CFGOPT="--with-pkgversion=GnuCOBOL-CI-MSYS2-debug --enable-debug --enable-cobc-internal-checks --enable-hardening" >> $env:GITHUB_ENV
+            echo CFGOPT="--with-pkgversion=GnuCOBOL-CI-MSYS2-Debug --enable-debug --enable-cobc-internal-checks --enable-hardening" >> $env:GITHUB_ENV
           }
           # Common options (dependencies)
           echo CFGOPT="$env:CFGOPT --with-math=gmp --with-curses=ncursesw --with-xml2 --with-json=json-c" >> $env:GITHUB_ENV
           # Conditional inclusion of --with-db (MSYS2 does not provide 32bit builds for BDB any more)
-          if ("${{ matrix.sys }}" -ne "mingw32") {
+          If ("${{ matrix.sys }}" -ne "mingw32") {
             echo CFGOPT="$env:CFGOPT --with-db" >> $env:GITHUB_ENV
-          } else {
+          } Else {
             echo CFGOPT="$env:CFGOPT --without-db" >> $env:GITHUB_ENV
           }
 
-      - name: Update and install packages
+      - name: Install packages
         uses: msys2/setup-msys2@v2
         with:
           update: true
-          msystem: ${{matrix.sys}}
-          install: make diffutils
-                   mingw-w64-${{matrix.env}}-cc
-                   mingw-w64-${{matrix.env}}-gmp
-                   mingw-w64-${{matrix.env}}-gettext-runtime
-                   mingw-w64-${{matrix.env}}-ncurses
-                   mingw-w64-${{matrix.env}}-libxml2
-                   mingw-w64-${{matrix.env}}-json-c
-
-      - name: BDB package
+          msystem: ${{ matrix.sys }}
+          install: autoconf automake make libtool flex bison
+                   mingw-w64-${{ matrix.env }}-cc
+                   mingw-w64-${{ matrix.env }}-gmp
+                   mingw-w64-${{ matrix.env }}-ncurses
+                   mingw-w64-${{ matrix.env }}-libxml2
+                   mingw-w64-${{ matrix.env }}-json-c
+
+      - name: Install BDB package
+        if: ${{ matrix.sys != 'mingw32' }}  # MSYS2 does not provide 32bit builds for it any more
         uses: msys2/setup-msys2@v2
-        if: matrix.sys != 'mingw32'  # MSYS2 does not provide 32bit builds for it any more
-        with:
-          msystem: ${{matrix.sys}}
-          install: mingw-w64-${{matrix.env}}-db
-
-      - name: Get CI dist tarball
-        uses: actions/download-artifact@v4
         with:
-          name: gnucobol-ci source distribution msys2
+          msystem: ${{ matrix.sys }}
+          install: mingw-w64-${{ matrix.env }}-db
 
       - name: Configure GnuCOBOL
         shell: msys2 {0}
         run: |
-          tar -xvf gnucobol*.tar.*  --strip-components=1
-          mkdir _build
-          cd _build
-          ../configure $CFGOPT
+          mkdir _build && cd _build
+          PATH="$PATH:/c/Program Files/Git/usr/bin" ../configure $CFGOPT CFLAGS=--std=c17
```

Comment:
we should add a note here why PATH is suffixed (... to allow configure finding a "diff" that works, to work around MSYS2 issues); note: I've checked on a fresh and updated MSYS2 installation and there

```sh
diff fx1 fx2 >/dev/null ; res0=$? ; diff fx1 fy >/dev/null ; res1=$?
echo $res0 $res1
```

resulted in the expected "0 1" output in each of msys, mingw64 and mingw32 shells?!?

Reply:
> resulted in the expected "0 1" output in each of msys, mingw64 and mingw32 shells?!?

Yes, as I mentionned in some other comment (don't remember where), diff works fine on a local MSYS2 installation. It also works fine in the "prepare" step of the workflow, but not in the further steps - and I spent hours trying to figure out why, to no avail.